auth.js

```diff
@@ -383,354 +373,360 @@ module.exports = function auth( options ) {
+          return next(null, {http$: {status: forceStatus || 301, redirect: forceRedirect || redirect.fail}})
         }
         else {
-          return next( null, {http$: {status: forceStatus || 200, redirect:forceRedirect}, ok: false, why: err} )
+          return next(null, {http$: {status: forceStatus || 200, redirect: forceRedirect}, ok: false, why: err})
         }
       }
-      else{
-        if( redirect ) {
-          req.seneca.log.debug( 'redirect', 'win', redirect.win )
-          return next( null, {http$: {status: forceStatus || 301,redirect:forceRedirect || redirect.win}} )
+      else {
+        if (redirect) {
+          req.seneca.log.debug('redirect', 'win', redirect.win)
+          return next(null, {http$: {status: forceStatus || 301, redirect: forceRedirect || redirect.win}})
         }
         else {
-          seneca.act( "role:'auth', cmd:'clean'", {user:req.seneca.user, login:req.seneca.login}, function( err, out ) {
+          seneca.act("role:'auth', cmd:'clean'", {user: req.seneca.user, login: req.seneca.login}, function (err, out) {
+            if (err) {
+              // handle error
+            }
+
             out.ok = true
-            out.http$ = { status: forceStatus || 200, redirect:forceRedirect }
-            return next( null, out )
+            out.http$ = {status: forceStatus || 200, redirect: forceRedirect}
+            return next(null, out)
           })
         }
       }
     })
   }
 
-
-  //LOGIN START
-  function afterlogin( err, next, req, res ) {
-    if( err ) {
-      return do_respond( err, 'login', req, next )
+  // LOGIN START
+  function afterlogin (err, next, req, res) {
+    if (err) {
+      return do_respond(err, 'login', req, next)
     }
 
-    if( req.user && req.user.ok ) {
+    if (req.user && req.user.ok) {
       // rename passport req.user prop
       req.seneca.user = req.user.user
       req.seneca.login = req.user.login
 
-      req.seneca.act( "role: 'auth', set: 'token'", {tokenkey: options.tokenkey, token: (
+      req.seneca.act("role: 'auth', set: 'token'", {
+        tokenkey: options.tokenkey, token: (
         req.seneca.login.id || req.seneca.login.id)
-      }, function( err ) {
-        return do_respond( err, 'login', req, next )
-      } )
+      }, function (err) {
+        return do_respond(err, 'login', req, next)
+      })
     }
     else {
-      do_respond( ( err ? err.why : 'Unknown error' ), 'login', req, next )
+      do_respond((err ? err.why : 'Unknown error'), 'login', req, next)
     }
   }
 
-
-  function cmd_login( msg, respond ) {
+  function cmd_login (msg, respond) {
     var req = msg.req$
     var res = msg.res$
 
-    req.query = _.extend( {}, req.query || {}, req.body || {} )
+    req.query = _.extend({}, req.query || {}, req.body || {})
+
+    seneca.act("role: 'auth', hook: 'map_fields'", {action: 'login', data: msg.data}, function (err, userData) {
+      if (err) {
+        // handle error
```

Comment:
i'm still not convinced about the error handling here, I will review the whole code handling error. Meanwhile I will continue with PR merging